### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by jaimevillegas**
Hi there!

@SFarida and I have made the following changes to the repository:

- Add devise authentication following project requirements

Thank you a lot for taking the time to review our code! 🚀 ✨ ⚡

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (jerryowusu) [inline app/models/user.rb:5]: It looks like the app is having issues with password confirmation when you try to create a new user. I tried creating a new user without the `:confirmable` and it worked fine. However, with `:confirmable`, it does not work but throws an error as shown below :point_down: 
Kindly fix it and make sure it works well with no errors. 

Screenshot from 2023-06-21 23-22-36

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi Team

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

so-close


## Highlights
Used devise gem :+1: 
Neat and well organized code :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (jaimevillegas) [reply app/models/user.rb:5]: Hello @jerryowusu 

Thank you so much to review our code!

We have worked on that. We've added a migration to add the column `confirmed_at` and other ones requested by confirmable. Now it works as expected. 🚀 ✨

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @jaimevillegas, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Image

 ## Required Changes ♻️
- [x] Kindly make sure to fix all errors which prevent the app from working properly. Currently, when I tried to run the migrations I got the following list of errors. You can simulate this by dropping, creating, and then setting up the database and you should get these errors.
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jaimevillegas** [comment]
Hello @ShadyShawkat 

Thank you for your comment!

We had an error with the migration, but now it's done. I have dropped and created all again, and migration runs without errors. Now I can enter `users/sign_up` and create an account or log in to access the blog app! ⚡

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,12 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  protect_from_forgery with: :exception
```
Comment (TadesseDev) [inline app/controllers/application_controller.rb:2]: - [x] You have done an incredible job adding the migration and fixing the errors. Thanks to your changes now it's possible to get to the login and sign-up pages. But the design of your `login` page is not following the wireframe provided. Please apply some stylings.

### Your design 
image

### Wireframe 
blog_login

- [Optional] I would suggest making your `Login` route your `root` path, this makes it easy for the user to sign in. I had to check out your routes to find the route for the login page.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @jaimevillegas 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (jaimevillegas) [reply app/controllers/application_controller.rb:2]: Hello @TadesseDev 

Thank you so much for your comments!

The design now matches the wireframe:
image

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @jaimevillegas, and @SFarida👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with authentication. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**